fastapi/routing.py

```diff
@@ -168,7 +168,10 @@ def get_request_handler(
     is_coroutine = asyncio.iscoroutinefunction(dependant.call)
     is_body_form = body_field and isinstance(body_field.field_info, params.Form)
     if isinstance(response_class, DefaultPlaceholder):
-        actual_response_class: Type[Response] = response_class.value
+        if status_code == HTTP_204_NO_CONTENT:
+            actual_response_class: Type[Response] = Response
```

Comment:
This seems to work as expected, even when you override the response status code, as documented here: https://fastapi.tiangolo.com/advanced/response-change-status-code/#response-change-status-code

I'm not opposed to this, but I wish we didn't have to add another conditional here, and it seems like we might get some unintended behavior, as we're really just taking advantage of the fact that Response converts `None` to an empty byte string, not that we're actually doing any validation of the result (I'm not proposing that we should be... either).

Reply:
I understand your concern, yet I didn't find any more gentle solution.. feel free to propose if you have any :)